### ISSUE TITLE
how to use it when using apex

### ISSUE THREAD

**Issue by ChenMinQi** [PhilJd/contiguous_pytorch_params#1]
when i try using contiguous params, it happens loss=NAN if i use apex O2 mode.
here is my code

```
parameters = ContiguousParams(network.parameters())
optimizer = torch.optim.SGD(parameters.contiguous())
network, optimizer = amp.initialize(network, optimizer, opt_level='O2')
network = torch.nn.parallel.DistributedDataParallel(network, device_ids=device_ids)
```

**Comment by PhilJd** [comment]
Yeah, I don't expect it to work with apex, as they replace the model weights with 16-bit versions, undoing any changes the contiguous params made. If you use Apex anyways, you should get the same speed up by using one of their fused optimizers!
